### ISSUE TITLE
repr(::Cmd) should not include control characters

### ISSUE THREAD

**Issue by tkelman** [JuliaLang/julia#22142]
https://github.com/JuliaLang/julia/pull/21619#discussion_r114020532
```
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.394 (2017-05-30 10:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 7c4edd3 (0 days old master)
|__/                   |  x86_64-linux-gnu

julia> repr(`foo "bar baz"`)
"`\e[4mfoo\e[24m \e[4m'bar baz'\e[24m`"
```

**Comment by stevengj** [comment]
I think the problem is that the [default `repr` function](https://github.com/JuliaLang/julia/blob/711291afee693fca8de5e60f5c4714bc0bf3be95/base/strings/io.jl#L139-L148) should disable color in its IOBuffer.

Or maybe color should be disabled automatically when printing to any IOBuffer?

**Comment by stevengj** [comment]
Hmm, but it also doesn't seem nice that the [default `show(io, ::Cmd)` method](https://github.com/JuliaLang/julia/blob/711291afee693fca8de5e60f5c4714bc0bf3be95/base/strings/io.jl#L139-L148) uses color.  I wouldn't want `print(somefile, cmd)` to use color control codes either.

**Comment by tkelman** [comment]
yes, only doing underlining in 3-arg show was the unaddressed review on #21619

**Comment by vtjnash** [comment]
dup https://github.com/JuliaLang/julia/issues/19682

**Comment by stevengj** [comment]
@vtjnash, this particular issue could be addressed by only doing underlining in the 3-arg `show` for `Cmd`, separate from #19682.
